operator/cilium_node.go

```diff
@@ -307,6 +307,65 @@ func (c *ciliumNodeUpdateImplementation) Update(origNode, node *cilium_v2.Cilium
 	return nil, nil
 }
 
+// CNPCallback is a generic operation to apply to a CiliumNetworkPolicy.
+type CNPCallback func(ctx context.Context, cnp *cilium_v2.CiliumNetworkPolicy) error
+
+// forEachCNP loops over each Cilium Network Policy, applying a callback to each one.
+// If clusterwide is true, then the selected policies will be CiliumClusterwideNetworkPolicy,
+// otherwise they will be CiliumNetworkPolicy.
+// limit defines how many network policies should be queried at once.
+func forEachCNP(ctx context.Context, clusterwide bool, limit int, cb CNPCallback) error {
+	continueID := ""
+
+	for {
+		var cnpItemsList []cilium_v2.CiliumNetworkPolicy
+
+		if clusterwide {
+			ccnpList, err := ciliumK8sClient.CiliumV2().CiliumClusterwideNetworkPolicies().List(
+				ctx,
+				meta_v1.ListOptions{
+					Limit:    int64(limit),
+					Continue: continueID,
+				})
+			if err != nil {
+				return err
+			}
+
+			cnpItemsList = make([]cilium_v2.CiliumNetworkPolicy, 0, len(ccnpList.Items))
+			for _, ccnp := range ccnpList.Items {
+				cnpItemsList = append(cnpItemsList, cilium_v2.CiliumNetworkPolicy{
+					Status: ccnp.Status,
```

Comment:
This operation won't be necessary for `ClearCnpStatusNodes` because we will remove the status. We will unnecessarily allocate more memory than the one we need just to basically store the CNP name for the `ClearCnpStatusNodes`. Sorry I also wanted to reuse `forEachCNP` but I guess it will need to be specific for the `ClearCnpStatusNodes` and for the `runCNPNodeStatusGC` use cases, unless we perhaps execute the `cb` here.

Reply:
Then I guess the best option is to completely separate the code paths and accept a little bit of code duplication.